pkg/agent/openflow/pipeline.go

```diff
@@ -1364,6 +1364,26 @@ func (f *featurePodConnectivity) l3FwdFlowsToRemoteViaTun(localGatewayMAC net.Ha
 	return flows
 }
 
+// l3FwdFlowFromGWToRemoteViaTun generates the flow to match the packets sourced from the Antrea gateway and destined for
+// remote Pods and forward them via tunnel. This flow is installed only in hybrid mode for matching reply packets of
+// Egress connections and these packets should be sent to remote Pods via tunnel.
+func (f *featurePodConnectivity) l3FwdFlowFromGWToRemoteViaTun(localGatewayMAC net.HardwareAddr, peerSubnet net.IPNet, tunnelPeer net.IP) binding.Flow {
+	ipProtocol := getIPProtocol(peerSubnet.IP)
+	flow := L3ForwardingTable.ofTable.BuildFlow(priorityNormal+1).
+		Cookie(f.cookieAllocator.Request(f.category).Raw()).
+		MatchProtocol(ipProtocol).
+		MatchRegMark(FromGatewayRegMark, FromExternalRegMark).                       // Match packets from external.
```

Comment:
why do we also need to match `FromGatewayRegMark` here? Wouldn't `MatchRegMark(FromExternalRegMark)` match the same packets?

Reply:
`FromGatewayRegMark` should be sufficient for reply Egress packets.